src/span.h

```diff
@@ -23,7 +23,12 @@ class Span
     constexpr Span(C* data, std::ptrdiff_t size) noexcept : m_data(data), m_size(size) {}
 
     constexpr C* data() const noexcept { return m_data; }
+    constexpr C* begin() const noexcept { return m_data; }
+    constexpr C* end() const noexcept { return m_data + m_size; }
     constexpr std::ptrdiff_t size() const noexcept { return m_size; }
+    constexpr C& operator[](std::ptrdiff_t pos) const noexcept { return m_data[pos]; }
```

Comment:
I think it's a bit dangerous to use `noexcept` for `operator[]`. If Span is used for a custom collection that might throw an exception here, using noexcept here will cause the app to terminate. `std::span`'s operator[] also does not use noexcept here either: http://en.cppreference.com/w/cpp/container/span/operator_at

Reply:
m_data is just a pointer; operator[] is pointer arithmetic + dereferencing of a pointer. There is no chance for anything custom.

Reply:
ah you are of course right